rules/no-array-for-each.js

```diff
@@ -407,22 +407,23 @@ const create = context => {
 					problem.fix = getFixFunction(node, functionInfo, context);
 				}
 
-				context.report(problem);
+				yield problem;
 			}
 		}
 	};
 };
 
-module.exports = {
-	create,
-	meta: {
-		type: 'suggestion',
-		docs: {
-			description: 'Prefer `for…of` over `Array#forEach(…)`.',
-			url: getDocumentationUrl(__filename)
-		},
-		fixable: 'code',
-		schema: [],
-		messages
+module.exports = createRule(
```

Comment:
It's unfortunate that this would prevent the [eslint-plugin-eslint-plugin](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin) lint rules from linting the rules.

Reply:
I understand that, but we can do things in `createRule` ourself. For example we don't need it enforce `schema`.

Reply:
Or, if we move this call to `index.js` and `test/`, the plugin can still recognize the rule.